ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -348,6 +378,307 @@ instance CardanoHardForkConstraints c => CanHardFork (CardanoEras c) where
+    -- 'composeTxOutTranslationPairs' to understand why this is partial but
+    -- is safe in the absence of Consensus bugs
+    translations ::
+      NP
+        (ApOnlySnd (K (CardanoTxOut c) -.-> TxOutWrapper))
+        (ShelleyBasedProtosAndEras c)
+    translations =
+        hmap
+          (\(ApOnlySnd f)
+             -> ApOnlySnd $ fn $ \(K (ShelleyTxOut x)) -> f `apFn` K (nsMapSnd x))
+          (composeTxOutTranslationPairs translateTxOut)
+
+-- Note that this is a HardForkBlock instance, but it's not compositional. This
+-- is because the LedgerTables relies on knowledge specific to Cardano and we
+-- have so far not found a pleasant way to express that compositionally.
+instance CardanoHardForkConstraints c => HasLedgerTables (LedgerState (CardanoBlock c)) where
+  -- TODO Right now, this definition corresponds to what is on the disk. It
+  -- might be possible to instead have this type be the Shelley table for the
+  -- latest era, ie a sum of maps instead of a map of sums. I'm not sure the
+  -- trade-offs.
+  newtype LedgerTables (LedgerState (CardanoBlock c)) mk = CardanoLedgerTables {
+        cardanoUTxOTable :: mk (SL.TxIn c) (CardanoTxOut c)
+      }
+    deriving (Generic)
+
+  projectLedgerTables (HardForkLedgerState hfstate) =
+      projectLedgerTablesHelper
+        (projectLedgerTables . unFlip)
+        hfstate
+
+  withLedgerTables (HardForkLedgerState hfstate) tables =
+        HardForkLedgerState
+      $ withLedgerTablesHelper
+          (\(Flip st) tables' -> Flip $ withLedgerTables st tables')
+          hfstate
+          tables
+
+  pureLedgerTables     f                                                                         = CardanoLedgerTables f
+  mapLedgerTables      f                                                 (CardanoLedgerTables x) = CardanoLedgerTables (f x)
+  traverseLedgerTables f                                                 (CardanoLedgerTables x) = CardanoLedgerTables <$> f x
+  zipLedgerTables      f                         (CardanoLedgerTables l) (CardanoLedgerTables r) = CardanoLedgerTables (f l r)
+  zipLedgerTables2     f (CardanoLedgerTables l) (CardanoLedgerTables c) (CardanoLedgerTables r) = CardanoLedgerTables (f l c r)
+  zipLedgerTablesA     f                         (CardanoLedgerTables l) (CardanoLedgerTables r) = CardanoLedgerTables <$> f l r
+  zipLedgerTables2A    f (CardanoLedgerTables l) (CardanoLedgerTables c) (CardanoLedgerTables r) = CardanoLedgerTables <$> f l c r
+  foldLedgerTables     f                                                 (CardanoLedgerTables x) = f x
+  foldLedgerTables2    f                         (CardanoLedgerTables l) (CardanoLedgerTables r) = f l r
+
+  namesLedgerTables = CardanoLedgerTables { cardanoUTxOTable = NameMK "cardanoUTxOTable" }
+
+instance CardanoHardForkConstraints c
+      => CanSerializeLedgerTables (LedgerState (CardanoBlock c)) where
+    codecLedgerTables = undefined
```

Comment:
We cannot currently define this. We will have to find out a different solution next week.